synapse/util/caches/expiringcache.py

```diff
@@ -55,11 +56,12 @@ def __init__(
         """
         self._cache_name = cache_name
 
+        self.max_size = int(max_len * cache_config.DEFAULT_CACHE_SIZE_FACTOR)
```

Comment:
Do we need this? It'll be set when we load the config anyway, and saves us from having to import it.

Reply:
Removed this line, though not sure what you're referring to when you say "load the config".

Reply:
This is needed by `test_global_instantiated_before_config_load` which presumes that the size of caches are scaled by DEFAULT_CACHE_SIZE_FACTOR before config load.

I don't know myself if that's something we want to do.

https://github.com/matrix-org/synapse/blob/905c83352d75e2810af21a641e3632f9687b0622/tests/config/test_cache.py#L97-L114

Reply:
Conclusion from #synapse-dev: yes, probably.